generators/client/templates/angular/src/main/webapp/app/shared/tracker/_tracker.service.ts

```diff
@@ -39,7 +37,7 @@ export class <%=jhiPrefixCapitalized%>TrackerService {
     connect () {
         if (this.connectedPromise === null) this.connection = this.createConnection();
         //building absolute path so that websocket doesnt fail when deploying with a context path
-        var loc = this.$window.location;
```

Comment:
we should either continue to use the`Window` object from ng2 or do something like http://juristr.com/blog/2016/09/ng2-get-window-ref/